### PR TITLE
Queue items

### PR DESCRIPTION
**Pull request by kevinhofmaenner**
I added some extra functionality to the delete action that reorders the queue items if a item is deleted. However, I cannot get the test to pass even though it is working flawlessly in the UI. Is there a problem with my test on line 139? Any help is greatly appreciated!

### CODE REVIEW

File: app/models/queue_item.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+
+  # delegate :title, to: :video, prefix: :video
+  #
+  # IS THE SAME THING AS
+  #
+  # def video_title
+  #   video.title
+  # end
```
Comment (matthewbarram) [inline app/models/queue_item.rb]: If you aren't using this IMO you should remove it.

File: app/controllers/queue_items_controller.rb [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
-  
+
   def index
-    @queue_items = [1,2]
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video])
+    queue_item = QueueItem.new(video: video, user: current_user, position: ((QueueItem.where(user: current_user).count))+1)
```
Comment (matthewbarram) [inline app/controllers/queue_items_controller.rb]: This line http://cl.ly/image/1K200A2W2g1u is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits

Comment (matthew-ta) [inline app/controllers/queue_items_controller.rb]: This line http://cl.ly/image/1h0y3g3f3E3z is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits

File: app/controllers/queue_items_controller.rb [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+      flash[:danger] = "Could not add video to Queue"
+      redirect_to video_path(video)
+    end
+  end
+
+  def destroy
+    queue_item = QueueItem.where(id: params[:id], user: current_user)
+
+    if !queue_item.empty? and queue_item.first.delete
+      current_user.queue_items.order(created_at: :asc).each_with_index do |qi, idx|
```
Comment (matthew-ta) [inline app/controllers/queue_items_controller.rb]: I would suggest using full words

File: app/models/queue_item.rb (hunk 1 above)
Comment (matthew-ta) [inline app/models/queue_item.rb]: If you aren't using I would suggest removing

File: spec/controllers/queue_items_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (matthew-ta) [inline spec/controllers/queue_items_controller_spec.rb]: remove extra lines here.

**Pull request merged**